server/src/main/java/de/uftos/utils/ConstraintInstanceDeleter.java

```diff
@@ -0,0 +1,62 @@
+package de.uftos.utils;
+
+import de.uftos.entities.ConstraintArgument;
+import de.uftos.entities.ConstraintInstance;
+import de.uftos.entities.ConstraintSignature;
+import de.uftos.repositories.database.ConstraintInstanceRepository;
+import de.uftos.repositories.database.ConstraintSignatureRepository;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.springframework.data.jpa.domain.Specification;
+
+/**
+ * A class that deletes all instances of constraints that contain a soon-to-be deleted entity
+ * as an argument.
+ */
+public class ConstraintInstanceDeleter {
+
+  private final ConstraintSignatureRepository constraintSignatureRepository;
+  private final ConstraintInstanceRepository constraintInstanceRepository;
+
+  public ConstraintInstanceDeleter(ConstraintSignatureRepository constraintSignatureRepository,
+                                   ConstraintInstanceRepository constraintInstanceRepository) {
+    this.constraintSignatureRepository = constraintSignatureRepository;
+    this.constraintInstanceRepository = constraintInstanceRepository;
+  }
+
+  /**
+   * Deletes all the instances containing an argument with these ids.
+   *
+   * @param ids the ids of the constraint arguments
+   */
+  public void removeAllInstancesWithArgumentValue(String[] ids) {
+    Specification<ConstraintSignature> signatureSpecificationBuilder =
+        new SpecificationBuilder<ConstraintSignature>()
+            .andDoubleJoinIn(ids, "instances", "arguments", "value")
+            .build();
+    List<ConstraintSignature> constraintSignatures =
+        constraintSignatureRepository.findAll(signatureSpecificationBuilder);
+
+    List<ConstraintInstance> constraintInstances = new ArrayList<>(constraintSignatures.stream()
+        .flatMap(constraintSignature -> constraintSignature.getInstances().stream()).toList());
+
+    List<String> studentIds = Arrays.stream(ids).toList();
```

Comment:
I guess "studentIds" is a result of copy-paste and not actually an accurate name here.
Probably "resourceIds" would be a better name.

Reply:
yes, thank you